harness/determined/ipc.py

```diff
@@ -133,36 +141,49 @@ def broadcast(self, obj: Any) -> None:
         self._pub_socket.send_pyobj(_SerialMessage(self._send_serial, obj))
         self._send_serial += 1
 
-    def gather_with_polling(self, health_check: Callable[[], None]) -> List[Any]:
+    def gather_with_polling(self, health_check: Callable[[], None]) -> Tuple[List[Any], bool]:
```

Comment:
Nit: This is a pretty weird function signature for python.  Why not just raise an exception here instead?  That would be more pythonic.

If you wanted to keep the BroadcastServer generic, then you could raise a `ConnectionError` here, and catch/translate it to a WorkerError later.  Same effect, but more pythonic.

Actually, looking further into it, you would necessarily have to distinguish it from `WorkerError` messages because we apparently don't respond to Terminate messages.  We should maybe do that so there's not the weird "Ignore WorkerError when we Terminated" case.  But that's obviously out of scope for this fix.